tests/pact/tc.repl

```diff
@@ -205,6 +205,29 @@
       (update persons "foo" partial)
       partial))
 
+  (defun fails-bind-bad-object:object{person} ()
+    (let ((o:object{person}
+            { 'name: 's
+            , 'age: 1
+            , 'dob: (parse-time "%F" "1996-12-31")
+            } ))
+      (bind o { 'name:= name, 'age:= age }
+        { 'foo: name
+        , 'bar: age })))
+
+  (defun fails-with-read-partial-schema:object{person} ()
+     (with-read persons 'dave
+       { 'name:= name
+       , 'age:= age }
+       { 'name: name
+       , 'age: age }))
+
+  (defcap CAP () true)
+
+  (defun fails-with-capability-bad-object:object{person} ()
```

Comment:
What is the reason for this test? Is this a regression test?

Reply:
yes, this regresses against a 3rd form of the bug Stuart found for capabilities as well. For this bug, we want to cover `bind`, `with-read`, and `with-capability`, which are the only forms that feature the bug.

Reply:
that makes sense

Reply:
well ... `with-default-read` isn't covered. I'll add a case